pysqldb3/tests/test_io.py

```diff
@@ -221,53 +221,200 @@
+        assert not sql.table_exists(table=test_pg_to_sql_qry_table)
+
+        # pg_to_sql_qry
+        data_io.pg_to_sql_qry(db, sql, query=
+                                f"""
+                                -- testing out comments
+                                select id, test_col1, test_col2 from /* what if there are comments here too */
+                                {pg_schema}.{pg_table_name} -- table name
+                                order by test_col1
+                                -- another comment
+                                limit 10; -- limit to 10 rows
+                                """,
+                              dest_table=test_pg_to_sql_qry_table,
+                              dest_schema = sql_schema,
+                              print_cmd=True)
+
+        # Assert sql to pg query was successful (table exists)
+        assert sql.table_exists(table=test_pg_to_sql_qry_table, schema = sql_schema)
+
+        # Assert df equality
+        pg_df = db.dfquery(f"""
+        select id, test_col1, test_col2 from {pg_schema}.{pg_table_name}
+        order by test_col1
+        limit 10
+        """).infer_objects().replace(r'\s+', '', regex=True)
+
+        # hardcoded the columns because they go in a different order when uploaded
+        sql_df = sql.dfquery(f"""
+        select id, test_col1, test_col2 from {sql_schema}.{test_pg_to_sql_qry_table}
+        order by test_col1
+        """).infer_objects().replace(r'\s+', '', regex=True)
+
+        # Assert
+        pd.testing.assert_frame_equal(pg_df, sql_df,
+                                    check_dtype=False,
+                                      check_column_type=False)
+
+        # Cleanup
+        sql.drop_table(schema=sql_schema, table=test_pg_to_sql_qry_table)
+
+    def test_pg_to_sql_qry_spatial(self):
+
+        sql.drop_table(schema=sql_schema, table=test_pg_to_sql_qry_spatial_table)
+        db.drop_table(schema = pg_schema, table = test_pg_to_sql_qry_spatial_table)
+
+        assert not db.table_exists(table=test_pg_to_sql_qry_spatial_table, schema = pg_schema)
+        assert not sql.table_exists(table=test_pg_to_sql_qry_spatial_table, schema = sql_schema)
+
+        # add new spatial table
+        db.query(f"""
+            create table {pg_schema}.{test_pg_to_sql_qry_spatial_table}
+                (test_col1 int, test_col2 int, test_geom geometry);
+            insert into {pg_schema}.{test_pg_to_sql_qry_spatial_table} (test_col1, test_col2, test_geom)
+                 VALUES (1, 2, ST_SetSRID(ST_MAKEPOINT(-71.1043443253471, 42.3150676015829), 2236));
+            insert into {pg_schema}.{test_pg_to_sql_qry_spatial_table} (test_col1, test_col2, test_geom)
+                VALUES (3, 4, ST_SetSRID(ST_MAKEPOINT(91.2763, 11.9434), 2236));
+        """)
+
+        data_io.pg_to_sql_qry(db, sql, query=f"""
+                                               SELECT test_col1, test_col2, ST_AsEWKT(test_geom) test_geom --comments within the query
```

Comment:
you should transfer it as regular geom but compare the results using wkt. Something like this 


data_io.pg_to_sql_qry(db, sql, query=f"""
                                               SELECT test_col1, test_col2, test_geom --comments within the query
                                                FROM {pg_schema}.{test_pg_to_sql_qry_spatial_table} -- geom here
                                                -- end here""",
                              dest_table= test_pg_to_sql_qry_spatial_table,
                              dest_schema = sql_schema,                      
                              spatial = True
                              )

pg_df = db.dfquery(f"""
        select test_col1, test_col2,  ST_Astext(est_geom) test_geom
        from {pg_schema}.{test_pg_to_sql_qry_spatial_table}
        order by test_col1
        """)
sql_df = sql.dfquery(f"""
                select test_col1, test_col2, test_geom.STAsText() as test_geom
                from {sql_schema}.{test_pg_to_sql_qry_spatial_table}
                order by test_col1
        """)